Sources/SwiftSyntaxMacros/MacroProtocols/BodyMacro.swift

```diff
@@ -0,0 +1,25 @@
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+/// Describes a macro that can create the body for a function that does not
+/// have one.
+public protocol BodyMacro: AttachedMacro {
+  /// Expand a macro described by the given custom attribute and
+  /// attached to the given declaration and evaluated within a
+  /// particular expansion context.
+  ///
+  /// The macro expansion can introduce a body for the given function.
+  static func expansion(
+    of node: AttributeSyntax,
+    providingBodyFor declaration: some DeclSyntaxProtocol & WithOptionalCodeBlockSyntax,
+    in context: some MacroExpansionContext
+  ) throws -> [CodeBlockItemSyntax]
```

Comment:
Wouldn’t it be nicer to return a `CodeBlockItemListSyntax` here instead of `[CodeBlockItemSyntax]`?

(same for `PreambleMacro`)

Reply:
The precent from other macros is to return an array rather than the `*ListSyntax` node.

Reply:
Is it, though? The only macro that I see that returns `[CodeBlockItemSyntax]` is the code item macro and that hasn’t been accepted, AFAICT, so it’s not really a precedence. And all the other macros return `[DeclSyntax]` or something more special that doesn’t have a corresponding list node in SwiftSyntax

Reply:
`MemberMacro` returns `[DeclSyntax]` instead of `MemberBlockItemSyntax`, for example.

Reply:
Well, they aren’t quite the same since `MemberBlockItemSyntax` can also contain a semicolon. 

And yes, you could make an argument that those should in theory return `MemberBlockItemListSyntax` as well.

Reply:
Let's chat either in the pitch thread or the review thread (when function body macros come up for a review).